drivers/sensor/lis2mdl/lis2mdl.c

```diff
@@ -351,12 +368,29 @@ static int lis2mdl_init(const struct device *dev)
 		return -EIO;
 	}
 
+#ifdef CONFIG_LIS2MDL_SINGLE_MODE
+	/* Set device in single mode */
+	if (lis2mdl_operating_mode_set(lis2mdl->ctx, LIS2MDL_SINGLE_TRIGGER)) {
+		LOG_DBG("set single mode failed\n");
+		return -EIO;
+	}
+
+	/* Set drdy on pin*/
+	if (lis2mdl_drdy_on_pin_set(lis2mdl->ctx, 1)) {
+		LOG_DBG("set drdy on pin failed!\n");
+		return -EIO;
+	}
```

Comment:
The setting of drdy_on_pin is already done calling lis2mdl_trigger_set() from API.
Shouldn't we do the same even in this case?

Reply:
The lis2mdl_trigger_set() will be called by the user if he/she wants to attach an **external** trigger to the interrupt. So the user is **free** to call lis2mdl_trigger_set() or **not** when it comes to "single-mode operation+interrupt". So we need to call the lis2mdl_drdy_on_pin_set() somewhere else to be sure that this is set since it is a crucial part of the "single-mode operation+interrupt".